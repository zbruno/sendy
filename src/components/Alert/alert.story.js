import React from 'react';

import Alert from '.';

export default {
  title: 'Alert',
  component: Alert,
  parameters: {
    jest: [],
  },
};

export const alerts = () => (
  <React.Fragment>
    <Alert
      title="Wow something went well!"
      type="success"
    >
      <Alert.Text>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus pellentesque interdum ipsum, ut feugiat libero iaculis nec. Vestibulum ut felis ante. In laoreet nulla quis ante porta pulvinar. Suspendisse vestibulum commodo dolor. Fusce at elit viverra, feugiat metus eu, varius lectus. Cras purus sapien, tristique eget dolor id, malesuada tincidunt velit. Nunc sit amet sapien tellus. Praesent vestibulum nulla vitae lorem malesuada, a vestibulum mi sagittis.
      </Alert.Text>
    </Alert>
    <Alert
      title="Something you should know..."
      type="info"
    />
    <Alert
      title="Oh boy. Not looking good."
      type="danger"
    />
  </React.Fragment>
);
