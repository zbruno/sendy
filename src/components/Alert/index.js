import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faCheckCircle,
  faTimesCircle,
  faInfoCircle,
} from '@fortawesome/free-solid-svg-icons';

const alertMapping = {
  success: { color: 'green', icon: faCheckCircle },
  danger: { color: 'red', icon: faTimesCircle },
  info: { color: 'teal', icon: faInfoCircle },
};

class Alert extends React.Component {
  static Container = ({ children }) => (
    <div className="alert-container">
      {children}
    </div>
  );

  static Text = ({ children }) => (
    <p className="alert-text">{children}</p>
  );

  static propTypes = {
    children: PropTypes.node,
    title: PropTypes.string.isRequired,
    type: PropTypes.oneOf(['success', 'danger', 'info']).isRequired,
  };

  static defaultProps = {
    children: null,
  };

  render() {
    const { children, type, title } = this.props;
    const alert = alertMapping[type];
    const alertClasses = cx({
      alert: true,
      [`bg-${alert.color}-100`]: true,
      [`border-${alert.color}-500`]: true,
    });

    return (
      <div className={alertClasses} role="alert">
        <div className={`inner-container text-${alert.color}-900`}>
          <FontAwesomeIcon
            icon={alert.icon}
            className={`alert-icon text-${alert.color}-500`}
            size="lg"
            fixedWidth
          />
          <span className="alert-text">{title}</span>
        </div>
        {children && (
          <div className={`alert-child text-${alert.color}-900`}>
            {children}
          </div>
        )}
      </div>
    );
  }
}

export default Alert;
