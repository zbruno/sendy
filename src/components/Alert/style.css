.alert-container {
  @apply flex items-end flex-col fixed p-6 inset-0 z-50 pointer-events-none;
}

.alert-text {
  @apply mb-4;

  &:last-of-type {
    @apply mb-0;
  }
}

.alert {
  @apply p-4 shadow mb-4 pointer-events-auto border-l-4;

  & .inner-container {
    @apply flex items-center leading-normal;
  }
}

.alert-icon {
  @apply fill-current mr-3;
}

.alert-text {
  @apply block font-semibold;
}

.alert-child {
  @apply text-sm leading-snug mt-2 opacity-75;
}
