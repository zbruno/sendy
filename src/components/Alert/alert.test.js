import React from 'react';
import renderer from 'react-test-renderer';

import Alert from './index';

const testAlert = {
  text: 'Your email address was updated successfully!',
  type: 'success',
  duration: 'sticky',
};

describe('Alert UI snapshots', () => {
  test('renders success type and text', () => {
    const component = renderer.create(
      <Alert
        title={testAlert.text}
        type="success"
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders danger type and text', () => {
    const component = renderer.create(
      <Alert
        title={testAlert.text}
        type="danger"
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders info type and text', () => {
    const component = renderer.create(
      <Alert
        title={testAlert.text}
        type="info"
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders container correctly', () => {
    const component = renderer.create(
      <Alert.Container />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders extra text', () => {
    const component = renderer.create(
      <Alert
        title={testAlert.text}
        type="info"
      >
        <Alert.Text>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus pellentesque interdum ipsum, ut feugiat libero iaculis nec. Vestibulum ut felis ante. In laoreet nulla quis ante porta pulvinar. Suspendisse vestibulum commodo dolor. Fusce at elit viverra, feugiat metus eu, varius lectus. Cras purus sapien, tristique eget dolor id, malesuada tincidunt velit. Nunc sit amet sapien tellus. Praesent vestibulum nulla vitae lorem malesuada, a vestibulum mi sagittis.</Alert.Text>
      </Alert>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
