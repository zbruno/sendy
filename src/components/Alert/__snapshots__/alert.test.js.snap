// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert UI snapshots renders container correctly 1`] = `
<div
  className="alert-container"
/>
`;

exports[`Alert UI snapshots renders danger type and text 1`] = `
<div
  className="alert bg-red-100 border-red-500"
  role="alert"
>
  <div
    className="inner-container text-red-900"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-times-circle fa-w-16 fa-fw fa-lg alert-icon text-red-500"
      data-icon="times-circle"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="alert-text"
    >
      Your email address was updated successfully!
    </span>
  </div>
</div>
`;

exports[`Alert UI snapshots renders extra text 1`] = `
<div
  className="alert bg-teal-100 border-teal-500"
  role="alert"
>
  <div
    className="inner-container text-teal-900"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-info-circle fa-w-16 fa-fw fa-lg alert-icon text-teal-500"
      data-icon="info-circle"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="alert-text"
    >
      Your email address was updated successfully!
    </span>
  </div>
  <div
    className="alert-child text-teal-900"
  >
    <p
      className="alert-text"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus pellentesque interdum ipsum, ut feugiat libero iaculis nec. Vestibulum ut felis ante. In laoreet nulla quis ante porta pulvinar. Suspendisse vestibulum commodo dolor. Fusce at elit viverra, feugiat metus eu, varius lectus. Cras purus sapien, tristique eget dolor id, malesuada tincidunt velit. Nunc sit amet sapien tellus. Praesent vestibulum nulla vitae lorem malesuada, a vestibulum mi sagittis.
    </p>
  </div>
</div>
`;

exports[`Alert UI snapshots renders info type and text 1`] = `
<div
  className="alert bg-teal-100 border-teal-500"
  role="alert"
>
  <div
    className="inner-container text-teal-900"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-info-circle fa-w-16 fa-fw fa-lg alert-icon text-teal-500"
      data-icon="info-circle"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="alert-text"
    >
      Your email address was updated successfully!
    </span>
  </div>
</div>
`;

exports[`Alert UI snapshots renders success type and text 1`] = `
<div
  className="alert bg-green-100 border-green-500"
  role="alert"
>
  <div
    className="inner-container text-green-900"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-check-circle fa-w-16 fa-fw fa-lg alert-icon text-green-500"
      data-icon="check-circle"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="alert-text"
    >
      Your email address was updated successfully!
    </span>
  </div>
</div>
`;
